zaza/model.py

```diff
@@ -1420,8 +1425,23 @@ def f(x):
     async def _unit_status():
         app = unit_name.split("/")[0]
         model_status = await async_get_status()
-        v = model_status.applications[app]['units'][unit_name][
-            'workload-status']['status']
+        if model_status.applications[app]['units']:
+            v = model_status.applications[app]['units'][unit_name][
+                'workload-status']['status']
+        else:
```

Comment:
It's generally more pythonic to do:

```python
try:
    v = model_status.applications[......]['status']
except KeyError:
    # now do the other thing when that wasn't available.
```

However, you may feel that the current implementation is more obvious?